cli/command/registry_test.go

```diff
@@ -0,0 +1,65 @@
+package command_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+
+	. "github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/internal/test"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/stretchr/testify/require"
+)
+
+type fakeClient struct {
+	client.Client
+	infoFunc func() (types.Info, error)
+}
+
+func (cli *fakeClient) Info(_ context.Context) (types.Info, error) {
+	if cli.infoFunc != nil {
+		return cli.infoFunc()
+	}
+	return types.Info{}, nil
+}
+
+func TestElectAuthServer(t *testing.T) {
+	testCases := []struct {
+		expectedAuthServer string
+		infoFunc           func() (types.Info, error)
+	}{
+		{
+			expectedAuthServer: "https://index.docker.io/v1/",
+			infoFunc: func() (types.Info, error) {
+				return types.Info{IndexServerAddress: "https://index.docker.io/v1/"}, nil
+			},
+		},
+		{
+			expectedAuthServer: "https://index.docker.io/v1/",
+			infoFunc: func() (types.Info, error) {
+				return types.Info{IndexServerAddress: ""}, nil
+			},
+		},
+		{
+			expectedAuthServer: "https://foo.bar",
+			infoFunc: func() (types.Info, error) {
+				return types.Info{IndexServerAddress: "https://foo.bar"}, nil
+			},
+		},
+		{
+			expectedAuthServer: "https://index.docker.io/v1/",
+			infoFunc: func() (types.Info, error) {
+				return types.Info{}, errors.Errorf("error getting info")
+			},
+		},
+	}
+	for _, tc := range testCases {
+		buf := new(bytes.Buffer)
+		cli := test.NewFakeCli(&fakeClient{infoFunc: tc.infoFunc}, buf)
+		server := ElectAuthServer(context.Background(), cli)
+		require.Equal(t, tc.expectedAuthServer, server)
```

Comment:
This is a good place to use `assert.Equal()` instead of `require` so that it will run all the test cases, instead of ending after the first failure.